### PR TITLE
docs: clarify rules docs

### LINKED ISSUE #780: Document error vs warning in RULES.md
**Is your feature request related to a problem? Please describe.**
We've documented whether all notices are errors or warnings in code. However, RULES.md doesn't say whether v2 considered each notice an error or warning. And confusingly, in the main table the v2 notices are mapped to v1 errors and warnings, but some of these have changed classification since v1 to more closely follow the spec, so the "E" and "W" in the v1 column can be misleading.

Plus, it's not easy to find a given notice in the current table, as there isn't any obvious sorting order.

**Proposed solution**
I propose that we move the current table in RULES.md to a new file MIGRATION_V1_V2.md, as it's primary purpose is to show how the notice names in v2 map to the codes in v1. We can add a link to the top of RULES.md to this new document.

In RULES.md, I propose that we have three tables:
* ERRORS
* WARNINGS
* INFO

In each we can have 3 columns - the notice name, the v2 notice code, and the title/explanation. Each table should be sorted alphabetically by notice name.

**Describe alternatives you've considered**
Leave documentation as-is.

### PR DESCRIPTION
**Pull request by lionel-nj**
closes #780 

**Summary:**

This PR updates the valdiator's rules documentation by:
- reorganizing `RULES.MD` in order to reflect the notice categorization 
- creating a new file `MIGRATION_V1_V2.md` to explain how notices were migrated from v1 to v2.
- 
**Expected behavior:** 

Nothing: no code change.

Please make sure these boxes are checked before submitting your pull request - thanks!

- [x] Run the unit tests with `gradle test` to make sure you didn't break anything
- [x] Format the title like "feat: [new feature short description]". Title must follow the Conventional Commit Specification(https://www.conventionalcommits.org/en/v1.0.0/).
- [x] Linked all relevant issues
- ~[ ] Include screenshot(s) showing how this pull request works and fixes the issue(s)~

### CODE REVIEW

File: docs/MIGRATION_V1_V2.md
Comment (barbeau) [reply docs/MIGRATION_V1_V2.md:5]: Also, given this is a migration document now, I'd change the labels for the columns to be "v1 Notice ID", "v2 Notice Name", and "v2 Notice category".

I guess it's debatable if we should replace the 2nd column with "notice code" instead of notice name, since they are the closest things that map to each other. But IMHO it's ok to just list the name as that's more readable and can hyperlink directly to class.

**Review by barbeau — COMMENTED** [review]
Thanks @lionel-nj! Some comments in-line.

File: docs/MIGRATION_V1_V2.md
Comment (lionel-nj) [reply docs/MIGRATION_V1_V2.md:13]: On it

File: RULES.md
Comment (lionel-nj) [reply RULES.md]: See https://github.com/MobilityData/gtfs-validator/pull/783

File: RULES.md
Comment (lionel-nj) [reply RULES.md]: > We should review all rule class names and codes to make sure they follow the same naming pattern (this could be done in another PR). They should be the same, but there are a few that are different like this example - SameNameAndDescriptionForRouteNotice and same_route_name_and_description.

Seehttps://github.com/MobilityData/gtfs-validator/pull/783

Comment (lionel-nj) [reply RULES.md]: > Also, I'd suggest we put the entity name first when it makes sense for consistency, and because if we list the rules in the table alphabetically then, for example, most of the Route rules should be grouped together. So, for example, this class and code could be renamed to RouteSameNameAndDescriptionNotice and route_same_name_and_description. Then it would appear right after RouteBothShortAndLongNameMissingNotice.

👍

File: docs/MIGRATION_V1_V2.md
Comment (lionel-nj) [reply docs/MIGRATION_V1_V2.md]: https://github.com/MobilityData/gtfs-validator/issues/785

**Comment by lionel-nj** [comment]
> [...] we don't even need to list the code in the table in RULES.md for #781


https://github.com/MobilityData/gtfs-validator/pull/781/commits/d44b3a8a5aa934634058ba3e87b3e078d40a42f5 removes codes from table as https://github.com/MobilityData/gtfs-validator/pull/783#issuecomment-789293904 suggests

File: RULES.md
Comment (lionel-nj) [reply RULES.md]: > Also, I'd suggest we put the entity name first when it makes sense for consistency, and because if we list the rules in the table alphabetically then, for example, most of the Route rules should be grouped together. So, for example, this class and code could be renamed to RouteSameNameAndDescriptionNotice and route_same_name_and_description. Then it would appear right after RouteBothShortAndLongNameMissingNotice.

Will be done in https://github.com/MobilityData/gtfs-validator/pull/783

File: RULES.md
Comment (barbeau) [inline RULES.md:229]: The above description could use an explanation of why the Apache Commons EmailValidator is listed here. I think there is one in the notice/validator? Maybe just copy/paste here?

Same with UrlValidator, and anything else that references outside libraries.

File: docs/MIGRATION_V1_V2.md
Comment (barbeau) [inline docs/MIGRATION_V1_V2.md:35]: To make it easier to see what wasn't implemented, I would leave the "Not implemented" text without a hyperlink, and then next to it add "(see pull request XXXXX)", with "pull request XXXXX" being hyperlinked.

File: docs/MIGRATION_V1_V2.md
Comment (barbeau) [inline docs/MIGRATION_V1_V2.md:63]: I would phrase this as "Not implemented (unnecessary with new architecture)"

File: docs/MIGRATION_V1_V2.md
Comment (barbeau) [inline docs/MIGRATION_V1_V2.md:64]: E060 and E061 are actually covered under `SystemErrors`, right?

**Review by barbeau — COMMENTED** [review]
Thanks @lionel-nj, this is looking good! A few issues to resolve and then I think it's done.

File: RULES.md
Comment (lionel-nj) [reply RULES.md:655]: Thanks for the trick! Updating this.

File: docs/MIGRATION_V1_V2.md
Comment (lionel-nj) [reply docs/MIGRATION_V1_V2.md:64]: Yes, modifying this. Thanks.

**Comment by lionel-nj** [comment]
This should be re-evaluated after merging https://github.com/MobilityData/gtfs-validator/pull/822/.

**Review by barbeau — APPROVED** [review]
Thanks @lionel-nj! This looks good. There are still a few small items that I think could be improved, but rather than to continue to go back and forth over a PR, let's get this merged so we can iterate on it from the master branch, given that other PRs will change it too.

**Pull request merged**
